OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -83,682 +83,807 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 			var dict = new Dictionary<string, Func<string, bool>>
 			{
-				{ "state",
-					s =>
-					{
-						var state = Session.ClientState.Invalid;
-						if (!Enum<Session.ClientState>.TryParse(s, false, out state))
-						{
-							server.SendOrderTo(conn, "Message", "Malformed state command");
-							return true;
-						}
+				{ "state", State(server, conn, client) },
```

Comment:
Can you please split this (much needed!) refactoring change into its own commit before the commit adding the new behaviour?